presto-main/src/main/java/io/prestosql/dispatcher/LocalDispatchQuery.java

```diff
@@ -190,39 +190,39 @@ public Duration getTotalCpuTime()
     {
         return tryGetQueryExecution()
                 .map(QueryExecution::getTotalCpuTime)
-                .orElse(new Duration(0, MILLISECONDS));
+                .orElseGet(() -> new Duration(0, MILLISECONDS));
     }
 
     @Override
     public DataSize getTotalMemoryReservation()
     {
         return tryGetQueryExecution()
                 .map(QueryExecution::getTotalMemoryReservation)
-                .orElse(new DataSize(0, BYTE));
+                .orElseGet(() -> new DataSize(0, BYTE));
     }
 
     @Override
     public DataSize getUserMemoryReservation()
     {
         return tryGetQueryExecution()
                 .map(QueryExecution::getUserMemoryReservation)
-                .orElse(new DataSize(0, BYTE));
+                .orElseGet(() -> new DataSize(0, BYTE));
     }
 
     @Override
     public BasicQueryInfo getBasicQueryInfo()
     {
         return tryGetQueryExecution()
                 .map(QueryExecution::getBasicQueryInfo)
-                .orElse(stateMachine.getBasicQueryInfo(Optional.empty()));
+                .orElseGet(() -> stateMachine.getBasicQueryInfo(Optional.empty()));
     }
 
     @Override
     public QueryInfo getFullQueryInfo()
     {
         return tryGetQueryExecution()
                 .map(QueryExecution::getQueryInfo)
-                .orElse(stateMachine.updateQueryInfo(Optional.empty()));
+                .orElseGet(() -> stateMachine.updateQueryInfo(Optional.empty()));
```

Comment:
When we arrive here for a FINISHED query, `stateMachine.updateQueryInfo(Optional.empty()` could produce final query info (state isDone() and output stage not provided). Thus we would populate finalQueryInfo.

Since this was called unconditionally, this could be called for a query that just became FINISHED, resulting in incorrect finalQueryInfo.

Reply:
> Since this was called unconditionally, this could be called for a query that just became FINISHED, resulting in incorrect finalQueryInfo.

Was it somehow visible in stats/UI?